lib/core/src/wallet.rs

```diff
@@ -162,4 +174,30 @@ impl OnchainWallet for LiquidOnchainWallet {
             .derive_priv(&Secp256k1::new(), &path)?;
         Ok(bip32_xpriv)
     }
+
+    fn sign_message(&self, message: &str) -> Result<String> {
+        let seed = self
+            .lwk_signer
+            .seed()
+            .ok_or(anyhow!("Could not get signer seed"))?;
+        let secp = Secp256k1::new();
+        let keypair = ExtendedPrivKey::new_master(self.config.network.into(), &seed)
+            .map_err(|e| anyhow!("Could not get signer keypair: {e}"))?
+            .to_keypair(&secp);
+        // Prefix and double hash message
+        let hashed_msg = sha256::Hash::hash(&[LN_MESSAGE_PREFIX, message.as_bytes()].concat());
+        let double_hashed_msg = Message::from(sha256::Hash::hash(&hashed_msg));
```

Comment:
I don't think we need to concatenate these messages, they can be hashed together in one pass using a `HashEngine`.
See last example on [bitcoin_hashes::commonly-used-operations](https://docs.rs/bitcoin_hashes/latest/bitcoin_hashes/index.html#commonly-used-operations).
```
let mut engine = sha256::HashEngine::default();
engine.input(LN_MESSAGE_PREFIX);
engine.input(message.as_bytes());
let double_hashed_msg = Message::from(sha256::Hash::from_engine(engine));
```
[write_all()](https://docs.rs/bitcoin_hashes/latest/bitcoin_hashes/sha256/struct.HashEngine.html#method.write) is preferred over `input()` if we want to handle errors early.